### DISCUSSION TITLE
What is the best way to power the Luxonis Oak D Lite ?

### DISCUSSION THREAD

**Discussion by jerrynsr** [iRobotEducation/create3_docs#127] [Questions] [answered]
Hi,
  I 3D Printed the Oak D Lite mount, got the camera installed.  I knew I would be pushing the PI4 power delivery limit but I still wanted to give it a try.  As soon as I executed "ros2 launch depthai_examples mobile_publisher.launch.py" I got a bus error due to the USB 3.0 power draw and reset the bus. Have any of you found a USBC power supply that can be used with the internal JST Create3 raw battery output connector ?    Is there any other solution ?

Here's a photo of GeorgeBot:
![georgebot-0822](https://user-images.githubusercontent.com/26339557/182687381-22a3df4f-e939-4c5c-a78c-71fc45ff87b9.jpg)

**Comment by shamlian** [comment, marked as answer]
I have not tried to power an Oak D yet. Luxonis has this FAQ: https://docs.luxonis.com/projects/hardware/en/latest/pages/guides/raspberrypi.html in which they look to sell you a wye for powering the camera. If I were to try to power the camera via a wye, I would get a 5V step-down (buck) converter intended for model cars or planes (also known as a [BEC](https://en.wikipedia.org/wiki/Battery_eliminator_circuit)), and wire it to a USB downstream breakout board. Georgebot looks great!!

**Comment by jerrynsr** [comment]
I bought the Oak D from Robotshop.   They didn't have the wye at the time.   I was looking at a 5v step down converter.   I wasn't sure if 3A is enough.   I'll check out the Luxonis faq again.

Thanks!

**Reply by shamlian** [reply]
Sure thing. FAQ seems to indicate 1 A is max.

**Comment by jerrynsr** [comment]
I just checked the power consumption of the ADDATA M.2 256GB SSD that I have in my Create3.  It's 760ma max. Add the 1 AMP max from the Oak D add 600ma max for the SLAMTEC A-1   and @300ma for the USB audio speaker,  I'm lucky the data on the SSD didn't get corrupted or worse.. a release of magic smoke from the RPI4 USB controller,  LOL.

**Comment by a7ypically** [comment]
I tried a 12v powered USB hub connected directly to the unregulated battery port of the Create 3. Seems to work fine though I did not test it for long.
I'm using this one:
https://www.amazon.com/gp/aw/d/B07BBLL3MJ

**Comment by jerrynsr** [comment]
I was going to solder up the supplied jst power cable to my USB power DC->DC converter.  Bummer.  I lost the 2 conductor JST cable.  What size JST 2 conductor pigtail do I need ?

**Reply by shamlian** [reply]
The information you seek is [here](https://iroboteducation.github.io/create3_docs/hw/adapter/#power-connections). It's a JST XH connector. Please, please double check polarity; you can buy pigtails from a number of retailers, but red and black might be swapped (this can be fixed by carefully removing and swapping the crimps from their housing, if necessary).

**Comment by jerrynsr** [comment]
I found the JST XH pigtail that was included with my bot.  I soldered up my  DC->USB 3.0 power supply.  The cables going to the camera will not clear the LIDAR head.   I ordered an usbc 1ft extension cable with a right angle connector on the end.
I had a spare standard usbc cable with right angle connector, it appeared to give enough clearance.  I will feed the extension cable down to the payload bay, attach the data/power t to the end, and connect the power and data connections in the payload bay.

The cable arrives tomorrow.

If this doesn't work the I'll 3D Print a base for the Oak D Lite camera to raise the mount

**Comment by jerrynsr** [comment]
Power is working great but the team that wrote the Communications link code dropped the ball.  USB3 High Speed does not work.
I tested with multiple cables, including a high quality 6" usb c cable.  Nada, Zippo.... after 30 seconds disconnect.  Apparently the team that wrote the code is aware of the buggy link and included a work around:
When creating the device pipeline add this after pipeline:
, usb2Mode=True

This is not a cable or USB 3.0 port problem.  I connected directly to my Linux desktop which is an Intel Haydes Canyon NUC.. it didn't work.  I connected it to my MacPro.. also same deal. Disconnect after 30 seconds.

With the device pipeline throttled down to usb 2.0 high speed, it works.
